src/app/core/models/image-viewer-layer.ts

```diff
@@ -39,4 +39,49 @@ export class ImageViewerLayer {
 
     return gradient;
   }
+
+}
+
+/**
+ * Converts a hex color code to RGB and calculates a value for the color brightness
+ * @param color Hex color code
+ * @returns brightness value
+ */
+function getBrightness(color: string): number {
```

Comment:
Move these into the ImageViewerLayer class. You'll have to update the constructor argument by adding these to the `Omit<..., 'background' | 'getBrightness' | ...>`